app/databaseAccessors/userAccessor.js

```diff
@@ -28,112 +28,110 @@ export default class UsersAccessor {
+ * Get a list of User IDs by a list emails
+ *
+ * @param {[String]} email - Array of email
+ * @returns {{[ObjectId]}} - Array of user ID
+ */
+  static async getUserIdsByMultipleEmails(emails) {
     const userIds = [];
-    for (const username of usernames) {
-      const user = await this.getUserByUsername(username);
+    for (const email of emails) {
+      const user = await this.getUserByEmail(email);
       if (user) {
         userIds.push(user._id);
       } else {
-        throw new ErrorUserNotFound(`User not found for username: ${username}`);
+        throw new ErrorUserNotFound(`User not found for email: ${email}`);
       }
     }
     return userIds;
   }
 
   /**
-   * getApprovedByUsername Method
+   * getApprovedByEmail Method
    *
    * This method retrieves the user MongoDB object from the
-   * database based on a given username
+   * database based on a given email
    *
-   * @param {String} username
-   * @returns the User associated with the given username in
+   * @param {String} email
+   * @returns the User associated with the given email in
    * the database.
    */
-  static async getApprovedByUsername(username) {
+  static async getApprovedByEmail(email) {
     await Connection.open();
     const user = await User.findOne({
-      username: username,
+      email: email,
       status: AccountStatus.Approved.toString(), // Use MongoDB filter for equal to approved status
     });
     return user;
   }
 
   /**
-   * getUnapprovedByUsername Method
+   * getUnapprovedByEmail Method
    *
    * This method retrieves the user MongoDB object from the
-   * database based on a given username
+   * database based on a given email
    *
-   * @param {String} username
-   * @returns the User associated with the given username in
+   * @param {String} email
+   * @returns the User associated with the given email in
    * the database.
    */
-  static async getUnapprovedByUsername(username) {
+  static async getUnapprovedByemail(email) {
     await Connection.open();
     const user = await User.findOne({
-      username: username,
+      email: email,
       status: { $ne: AccountStatus.Approved.toString() }, // Use MongoDB filter for not equal to approved status
     });
     return user;
   }
 
   /**
-   * approveUserByUsername Method
+   * approveUserByEmail Method
    *
    * This method retrieves the user MongoDB object from the
-   * database based on a given username and update the user's status to approved.
+   * database based on a given email and update the user's status to approved.
    *
-   * @param {String} username
+   * @param {String} email
    * @returns the updated user object with the status set to approved.
    */
-  static async approveUserByUsername(username) {
+  static async approveUserByEmail(email) {
     await Connection.open();
     //update the status
     const user = await User.findOneAndUpdate(
-      { username: username },
+      { emails: email },
```

Comment:
`email` field on user is singular in this PR
```suggestion
      { email: email },
```